File: UI_React/src/app/services/randomDescriptionService.ts
const definedNegativeDescriptions: ReadonlyArray<string> = Object.freeze([
    "Groceries supermarket (React)",
    "Dining (React)",
    "Utilities (React)",
    "Rent (React)",
    "Transportation (React)",
    "Health insurance (React)",
    "Entertainment (React)",
    "Tuition (React)",
    "Clothing (React)",
    "Savings (React)",
    "Travel (React)",
    "Gifts donations (React)",
    "Subscriptions (React)",
    "Personal care (React)",
    "Home maintenance (React)",
    "Pet supplies (React)",
    "Electronics (React)",
    "Sports fitness (React)",
    "Books (React)",
    "Childcare (React)"
]);

const definedPositiveDescriptions: ReadonlyArray<string> = Object.freeze([
    "Salary (React)",
    "Freelance work (React)",
    "Investments (React)",
    "Dividends (React)",
    "Interest income (React)",
    "Rental income (React)",
    "Business profits (React)",
    "Bonuses (React)",
    "Royalties (React)",
    "Part-time job (React)"
]);

const RandomDescriptionService = {
      getRandomPositiveDescription(): string {
        const randomIndex = Math.round(Math.random() * definedPositiveDescriptions.length)
        const randomDescription = definedPositiveDescriptions[randomIndex];

        return randomDescription;
      },
      getRandomNegativeDescription(): string {
        const randomIndex = Math.round(Math.random() * definedNegativeDescriptions.length)
        const randomDescription = definedNegativeDescriptions[randomIndex];

        return randomDescription;
      }
}

export default RandomDescriptionService;

File: UI_Blazor_WebApp/UI_Blazor_WebApp/Services/RandomDescriptionService.cs
namespace UI_Blazor_WebApp.Services
{
    public class RandomDescriptionService
    {
        private List<string> definedNegativeDescriptions = new List<string> {
            "Groceries supermarket (Blazor)",
            "Dining (Blazor)",
            "Utilities (Blazor)",
            "Rent (Blazor)",
            "Transportation (Blazor)",
            "Health insurance (Blazor)",
            "Entertainment (Blazor)",
            "Tuition (Blazor)",
            "Clothing (Blazor)",
            "Savings (Blazor)",
            "Travel (Blazor)",
            "Gifts donations (Blazor)",
            "Subscriptions (Blazor)",
            "Personal care (Blazor)",
            "Home maintenance (Blazor)",
            "Pet supplies (Blazor)",
            "Electronics (Blazor)",
            "Sports fitness (Blazor)",
            "Books (Blazor)",
            "Childcare (Blazor)"
        };

        private List<string> definedPositiveDescriptions = new List<string>
        {
            "Salary (Blazor)",
            "Freelance work (Blazor)",
            "Investments (Blazor)",
            "Dividends (Blazor)",
            "Interest income (Blazor)",
            "Rental income (Blazor)",
            "Business profits (Blazor)",
            "Bonuses (Blazor)",
            "Royalties (Blazor)",
            "Part-time job (Blazor)"
        };

        public string GetRandomPositiveDescription()
        {
            var randomIndex = new Random().Next(0, definedPositiveDescriptions.Count);
            var randomDescription = definedPositiveDescriptions[randomIndex];

            return randomDescription;
        }

        public string GetRandomNegativeDescription()
        {
            var randomIndex = new Random().Next(0, definedNegativeDescriptions.Count);
            var randomDescription = definedNegativeDescriptions[randomIndex];

            return randomDescription;
        }
    }
}
